### PR TITLE
Custom logs (new implementation)

### PR DESCRIPTION
**Pull request by alex-schneider**
add `custom_log_fields` attribute to blocks be able to describe a user defined map for `custom` log field enrichment.

---
<details>
    <summary>Reviewer checklist</summary>
    <ul>
        <li>Read PR description: a summary about the changes is required</li>
        <li>Changelog updated</li>
        <li>Documentation: docs/{Reference, Cli, ...}, Docker and cli help/usage</li>
        <li>Pulled branch, manually tested</li>
        <li>Verified requirements are met</li>
        <li>Reviewed the code</li>
        <li>Reviewed the related tests</li>
    </ul>
</details>

### CODE REVIEW

File: logging/hooks/custom_logs.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+type CustomLogs struct{}
+
+func (c *CustomLogs) Levels() []logrus.Level {
+	return logrus.AllLevels
+}
+
+func (c *CustomLogs) Fire(entry *logrus.Entry) error {
+	if entry.Context != nil {
+		if t, exists := entry.Data["type"]; exists {
+			switch t {
```
Comment (malud) [inline logging/hooks/custom_logs.go:23]: Both `type`'s are configurable. We have to pass the logging configuration to this hook.

Comment (alex-schneider) [reply logging/hooks/custom_logs.go:42]: Done

File: handler/error.go [hunk 2]
```diff
@@ -37,12 +37,15 @@ func (e *Error) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	}
 
 	for _, kind := range err.Kinds() {
+		if kind == "access_control" {
```
Comment (malud) [inline handler/error.go:40]: why override a local var ?
A better place would be the fallback wildcard lookup at line 61. Which means the Interface lookup on the endpoint handler may turn into a method.

Comment (malud) [reply handler/error.go]: refactored

**Review by malud — APPROVED** [review]

**Pull request merged**